optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/acceptor/greatdeluge/GreatDelugeAcceptor.java

```diff
@@ -0,0 +1,113 @@
+package org.optaplanner.core.impl.localsearch.decider.acceptor.greatdeluge;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.impl.localsearch.decider.acceptor.AbstractAcceptor;
+import org.optaplanner.core.impl.localsearch.scope.LocalSearchMoveScope;
+import org.optaplanner.core.impl.localsearch.scope.LocalSearchPhaseScope;
+import org.optaplanner.core.impl.localsearch.scope.LocalSearchStepScope;
+import org.optaplanner.core.impl.score.ScoreUtils;
+
+public class GreatDelugeAcceptor extends AbstractAcceptor {
+
+    private Score initialLevel = null;
+
+    private Double rainSpeed = null;
+    private Double rainSpeedRatio = null;
+
+    private int levelsLength = -1;
+    private double[] initialLevelScoreLevels;
+    private double[] levelScoreLevels;
+
+    private double levelMinimum = 0;
+    private static final double THRESHOLD = .0000001;
+
+    // Good value to come out from. Source: https://github.com/UniTime/cpsolver from Tomas Muller
+    private static final double DEFAULT_RAIN_SPEED_RATIO = 0.99_999_995;
+
+    public void setInitialLevels(Score initialLevel) { this.initialLevel = initialLevel; }
+
+    public Score getInitialLevels() { return this.initialLevel; }
+
+    public void setRainSpeed(Double rainSpeed) { this.rainSpeed = rainSpeed; }
+
+    public Double getRainSpeed() { return this.rainSpeed; }
+
+    public void setRainSpeedRatio(Double rainSpeedRatio) {this.rainSpeedRatio = rainSpeedRatio; }
+
+    public Double getRainSpeedRatio() { return this.rainSpeedRatio; }
+
+    public void phaseStarted(LocalSearchPhaseScope phaseScope) {
+        super.phaseStarted(phaseScope);
+
+        if (initialLevel != null) {
+            for (double initialLevelLevel : ScoreUtils.extractLevelDoubles(initialLevel)) {
+                if (initialLevelLevel < 0.0) {
+                    throw new IllegalArgumentException("The initial level (" + initialLevel
+                            + ") cannot have negative level (" + initialLevelLevel + ").");
+                }
+            }
+            initialLevelScoreLevels = ScoreUtils.extractLevelDoubles(initialLevel);
+            levelScoreLevels = initialLevelScoreLevels;
+        } else {
```

Comment:
This branch contains juicy logic which hasn't been covered by any test yet.

Reply:
Everything should be covered by tests now.

Reply:
thanks